packages/block-library/src/post-excerpt/edit.js

```diff
@@ -69,6 +69,34 @@ export default function PostExcerptEditor( {
 			[ `has-text-align-${ textAlign }` ]: textAlign,
 		} ),
 	} );
+	/**
+	 * Some themes might use `excerpt_more` filter to add links
+	 * to excerpts generated from content. Since we display the
+	 * renderedExcerpt from REST API links might be included.
+	 * Before the application of this filter all content has been
+	 * stripped of tags, so if we find any links there should come
+	 * from the filter.
+	 *
+	 * In order to avoid a possible `inception` effect
+	 * see: (https://github.com/WordPress/gutenberg/issues/33309)
+	 * we change the `href` attribute of links to a `pseudo link`.
+	 *
+	 */
+	const filteredRenderedExcerpt = useMemo( () => {
+		const document = new window.DOMParser().parseFromString(
+			renderedExcerpt,
+			'text/html'
+		);
+		const links = document.getElementsByTagName( 'a' );
+		if ( ! links?.length ) {
+			return renderedExcerpt;
+		}
+		for ( const link of links ) {
+			link.setAttribute( 'href', '#excerpt_more-pseudo-link' );
+		}
+		return document.body.innerHTML || '';
+	}, [ renderedExcerpt ] );
```

Comment:
What about just wrapping the content with `Disabled` component like we do in several other blocks? Is that enough?

Reply:
Oh.., I guess this could do the trick for the `click link` problem in a more elegant way. I'll try it out.